cmd/kube-proxy/app/server.go

```diff
@@ -252,7 +253,33 @@ func NewProxyServerDefault(config *ProxyServerConfig) (*ProxyServer, error) {
 		glog.V(2).Info("Using iptables Proxier.")
 		proxierIptables, err := iptables.NewProxier(iptInterface, execer, config.IptablesSyncPeriod, config.MasqueradeAll)
 		if err != nil {
-			glog.Fatalf("Unable to create proxier: %v", err)
+			iptables_err := err.Error()
+			iptables_err_str := ""
+			if strings.Contains(iptables_err, "iptables") {
+				//	Error is most probably caused by lack of iptables
+				// configuration
+				iptables_err_str = "Check if iptables has been enabled in the kernel. " +
```

Comment:
I do agree with your observation. The fix discussed in issue #12459 was in the v1.0.2 time frame when we primarily had user-space proxy. I myself have been able to test it only v1.0.6 and I think the iptables-base proxier is getting introduced in v1.1. Wasn't able to test it against v1.1 or the HEAD since I am unable to build hyperkube containers. I checked @brendandburns  on google groups and he mentioned that a new container image needs to be cut for v1.1 which is still in the works.

Going back to your proposal, it does seem a bit intrusive. What I can do is go through proxier.go and see where the iptables commands are getting executed and add this warning to which ever command fails. On looking at the iptables proxier code, the iptables should getting executed in the syncProxyRules function. However, wasn't able to find an implementation of this function anywehre ?? A Noob to golang so could be missing something here.